lib/dns.js

```diff
@@ -38,6 +37,9 @@ function errnoException(err, syscall, hostname) {
   return ex;
 }
 
+function validIpVersion(isIpResult) {
+  return isIpResult !== 0;
```

Comment:
IMHO I don't think it's necessary to extract a conditional like this into its own function.